logger.go

```diff
@@ -25,17 +25,33 @@ import (
 )
 
 var (
-	infoLog     *log.Logger
-	errorLog    *log.Logger
-	fatalLog    *log.Logger
-	logLock     sync.Mutex
-	initialized bool
+	defaultLogger *logger
+	initialized   bool
+	logLock       sync.Mutex
 )
 
+const flags = log.Ldate | log.Lmicroseconds | log.Lshortfile
+
+func initialize() {
+	initText := "ERROR: Logging before logger.Init.\n"
+	defaultLogger = &logger{
+		infoLog:  log.New(os.Stderr, initText+"INFO: ", flags),
+		errorLog: log.New(os.Stderr, initText+"ERROR: ", flags),
+		fatalLog: log.New(os.Stderr, initText+"FATAL: ", flags),
+	}
+}
+
+func init() {
+	initialize()
```

Comment:
That would require the tests to run in a specific order, or rewrite the initialization in the tes, I felt this was cleaner.